main.c

```diff
@@ -169,12 +169,26 @@ static void handle_selection_start(struct slurp_seat *seat,
 	struct slurp_state *state = seat->state;
 	current_selection->has_selection = true;
 	
-	if (state->single_point) {
-		state->result.x = current_selection->x;
-		state->result.y = current_selection->y;
-		state->result.width = state->result.height = 1;
+	if (state->single_point || state->output_selection) {
+        
+        struct slurp_output *output;
+        wl_list_for_each(output, &state->outputs, link) {
+            if(in_box(&output->logical_geometry, current_selection->x, current_selection->y)) {
+                break;
+            }
+        }
+        state->result.output = output;
```

Comment:
Would it make sense to set this for a rectangular selection as well? You do run into the problem of what to do if there are multiple outputs, but the simplest think would be just to pick a point, like the top left, or center of the rectangle, and use that to pick the "best" output.

Reply:
Slurp can take a predefined set of rectangles as input and limit possible selections to those rectangles (you can see this if you use the output selection example in the README).

I think the correct approach is to implement it this way, where there's no option to select anything other than the predefined rectangles which occupy each output entirely, and chosen by the rectangle the mouse is inside of.

Reply:
Looks like you can just use `add_choice_box` like is done here https://github.com/emersion/slurp/blob/master/main.c#L761

Reply:
> limit possible selections to those rectangles

That isn't quite right. At least from me trying it out in slurp 1.2.0, it seems it just creates predefined boxes you can select with a click, but it doesn't stop you from dragging your own box.

Reply:
Hmmm, that sounds like a bug in the intended behavior, but @emersion would have to weigh in.

Reply:
No, this is intentional: the input boxes are just hints, the user can ignore the hint. I'd be fine with adding an option to prevent that.

Reply:
Glad I asked.